core/src/main/java/com/sequenceiq/cloudbreak/controller/ClusterCreationSetupService.java

```diff
@@ -169,16 +176,21 @@ public Cluster prepare(ClusterRequest request, Stack stack, Blueprint blueprint,
                 StackRepoDetails stackRepoDetails = conversionService.convert(ambariStackDetails, StackRepoDetails.class);
                 stackRepoDetailsJson = new Json(stackRepoDetails);
             } else {
-                StackRepoDetails repo = defaultHDPInfo(blueprint, request, user).getRepo();
-                pruneVDFUrlsByOsType(cluster, stackImageComponent, repo);
-                stackRepoDetailsJson = new Json(repo);
+                StackRepoDetails stackRepoDetails = defaultHDPInfo(blueprint, request, user).getRepo();
+                Optional<String> vdfUrl = getVDFUrlByOsType(stackId, stackRepoDetails);
+                vdfUrl.ifPresent(s -> stackRepoDetails.getStack().put(CUSTOM_VDF_REPO_KEY, s));
+                stackRepoDetailsJson = new Json(stackRepoDetails);
             }
         } else {
             stackRepoDetailsJson = stackHdpRepoConfig.get().getAttributes();
         }
-        ClusterComponent hdpRepoComponent = new ClusterComponent(ComponentType.HDP_REPO_DETAILS, stackRepoDetailsJson, cluster);
-        components.add(hdpRepoComponent);
-        return components;
+
+        AmbariRepo ambariRepo = conversionService.convert(request.getAmbariRepoDetailsJson(), AmbariRepo.class);
+        if (ambariRepositoryVersionService.isNewAmbariApi(ambariRepo) && !containsVDFUrl(stackRepoDetailsJson)) {
+            throw new BadRequestException("Couldn't determine any VDF file");
```

Comment:
It would be nice if the error message would contain the os-type, os and Ambari version.